docs/spec/v1beta1/provider.md

```diff
@@ -215,6 +215,70 @@ kubectl create secret generic tls-certs \
   --from-file=caFile=ca.crt
 ```
 
+### Slack App
+
+It is possible to use a Slack App bot integration to send messages. To obtain a bot token, follow
+[Slack's guide on bot users](https://api.slack.com/bot-users).
+
+Differences from the Slack [webhook method](#notifications):
+
+* Possible to use single credentials to post to different channels (by adding the integration to each channel)
+* All messages are posted with the app username, and not the name of the controller (e.g. `helm-controller`, `source-controller`)
+
+To enable the Slack App, the secret must contain the URL of the [chat.postMessage](https://api.slack.com/methods/chat.postMessage)
+method and your Slack bot token (starts with `xoxb-`):
+
+```shell
+kubectl create secret generic slack-token \
+--from-literal=address=https://slack.com/api/chat.postMessage \
+--from-literal=token=xoxb-YOUR-TOKEN
+```
+
+Then reference this secret in `spec.secretRef`:
+
+```yaml
+apiVersion: notification.toolkit.fluxcd.io/v1beta1
+kind: Provider
+metadata:
+  name: slack
+  namespace: default
+spec:
+  type: slack
+  channel: general
+  # HTTP(S) proxy (optional)
+  proxy: https://proxy.corp:8080
+  # secret containing Slack API address and token
+  secretRef:
+    name: slack-token
+```
+
+### MS Teams
+
+Create an incoming webhook on the Microsoft Teams UI:
+
+1. Open the settings of the channel you want the notifications to be sent to.
+2. Click on `Connectors`.
+3. Click on the `Add` button for Incoming Webhook.
+4. Click on Configure and copy the webhook url given.
+
```

Comment:
@somtochiama 
>  Are there any other parts that you think are missing?

Right, so this step-by-step was replaced with links in [the new doc](https://fluxcd.io/flux/components/notification/provider/#microsoft-teams-example):

> create a Secret with [the address](https://fluxcd.io/flux/components/notification/provider/#address-example) set to the [webhook URL](https://learn.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/add-incoming-webhook#create-incoming-webhooks-1)

The only suggestion I could have is to keep the step by step form as the current single sentence "To configure a..." is not quite explicit, especially to newbies like myself. So, I would go for something like this:

1. Create an [incoming webhook](https://learn.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/add-incoming-webhook?tabs=dotnet#create-incoming-webhooks-1) on the Microsoft Teams and copy the webhook URL.
2. Create a Secret with [the address](https://fluxcd.io/flux/components/notification/provider/#address-example) set to the webhook URL.
3. Create a Provider for Microsoft Teams with a [Secret reference](https://fluxcd.io/flux/components/notification/provider/#address-example):
    ```
    ---
    apiVersion: notification.toolkit.fluxcd.io/v1beta2
    kind: Provider
    ...
    ```